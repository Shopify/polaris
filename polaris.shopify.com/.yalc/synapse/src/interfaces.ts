import type { ChatCompletionRequestMessage } from "openai";
import type { Node, Parent, Data } from "unist";
import type { Plugin } from "unified";

export interface Bit {
  title: string;
  text: string;
  embedding: Vector;
  tokenCount: number;
  slug?: string;
}

// a wrapper for Bit that unified can use
export interface BitNode extends Node {
  type: "BitNode";
  data: Data & Bit;
}

export interface SynapseTree extends Parent {
  type: "root";
  children: BitNode[];
}

export type Vector = number[];

export type TemplateArgs = { input: string } & {
  [key: string]: any;
};

export type PromptTemplate = (
  args: TemplateArgs,
  similarBits?: Bit[]
) => string;

export type Message = ChatCompletionRequestMessage;

export type MessagesTemplate = (
  args: TemplateArgs,
  similarBits?: Bit[]
) => Message[];

export type SynapseOptions = {
  name: string;
  source: string[];
  // can override the shared outputDir
  outputDir?: string;
  plugins: (Plugin | [Plugin, {}])[];
};

export type SynapseConfig = {
  outputDir: string;
  synapses: SynapseOptions[];
};
