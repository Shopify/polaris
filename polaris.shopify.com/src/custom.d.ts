import 'react';

declare module 'react' {
  interface StyleHTMLAttributes<T> extends React.HTMLAttributes<T> {
    jsx?: boolean;
    global?: boolean;
  }
}
