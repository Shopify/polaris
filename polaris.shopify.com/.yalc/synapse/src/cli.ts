import { unified } from "unified";
import { VFile } from "vfile";
import path from "path";
import fs from "fs";
import fastGlob from "fast-glob";

import type { SynapseConfig, SynapseOptions } from "./synapse";

import { Command } from "commander";

const cwd = process.cwd();
const config = await loadSynapseConfig(cwd);

const program = new Command();

program
  .name("synapse")
  .description("CLI for running Synapse commands")
  .version("1.0.0");

program
  .command("embeddings")
  .description("Transform raw data into Synapse bits")
  .option("--name [names...]", "names of the embeddings groups to run")
  .action((args) => {
    let synapses = config.synapses;
    const synapseNames = synapses.map((synapse) => synapse.name);

    if (args.name) {
      args.name.forEach((name: string) => {
        if (!synapseNames.includes(name)) {
          throw `"${name}" not found in your synapse.config.mts. Found: ${synapseNames.join(
            ", "
          )}`;
        }
      });

      // run just the synapes specified
      synapses = synapses.filter((synapse) => args.name.includes(synapse.name));
    }

    synapses.forEach(processSynapse);
  });

async function processSynapse(options: SynapseOptions) {
  // allow a synapse to override top level config options
  const { synapses, ...sharedConfigs } = config;

  options = {
    ...sharedConfigs,
    ...options,
  };

  const files = fastGlob.sync(options.source, {});

  // configure the unified processor
  const processor = unified();
  for (let plugin of options.plugins) {
    let params;
    // plugins can be configured with params
    if (Array.isArray(plugin)) {
      const [pluginFn, ...pluginParams] = plugin;
      plugin = pluginFn;
      [params] = pluginParams;
    }

    processor.use(plugin, params);
  }

  // read in the source files
  const vFiles = files.map((file) => {
    return new VFile({
      path: file,
      value: fs.readFileSync(file, "utf8"),
    });
  });

  // process and output them
  for (let vFile of vFiles) {
    const doc = await processor.process(vFile);
    const outPath = path.join(
      options.outputDir,
      path.basename(vFile.path, ".txt") + ".json"
    );
    fs.writeFileSync(outPath, doc.value, "utf8");
    console.log(`${outPath} generated.`);
  }
}

export async function loadSynapseConfig(cwd: string): Promise<SynapseConfig> {
  const configPath = "../../../synapse.config.mts";
  const configImport = await import(configPath);
  if (configImport) {
    return configImport.default;
  } else {
    console.warn("could not find synapse.config.mts");
  }
}

program.parse(process.argv);
