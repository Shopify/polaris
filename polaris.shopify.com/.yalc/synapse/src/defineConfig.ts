import type { SynapseConfig } from "./interfaces";

export const defineConfig = (options: SynapseConfig) => {
  return options;
};
