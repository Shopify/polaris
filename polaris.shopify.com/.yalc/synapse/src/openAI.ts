/**
 * utilities for working with OpenAI
 */

import GPT3TokenizerImport from "gpt3-tokenizer";
import type {
  Vector,
  Bit,
  PromptTemplate,
  TemplateArgs,
  MessagesTemplate,
  Message,
} from "./interfaces";
import { Configuration, OpenAIApi } from "openai";

// have to tweak the import of gpt3-tokenizer lib to make it work in node
const GPT3Tokenizer: typeof GPT3TokenizerImport =
  typeof GPT3TokenizerImport === "function"
    ? GPT3TokenizerImport
    : (GPT3TokenizerImport as any).default;

const DEFAULT_TOKEN_COUNT = 3000;
const MAX_COMPLETION_TOKENS = 600;

const openai = new OpenAIApi(
  new Configuration({
    apiKey: process.env.OPENAI_API_SECRET_KEY,
  })
);

export function encodeEmbedding(data: Vector) {
  return Buffer.from(new Float32Array(data).buffer).toString("base64");
}

export function decodeEmbedding(data: string) {
  return Array.from(
    new Float32Array(new Uint8Array(Buffer.from(data, "base64")).buffer)
  );
}

export function getTokenCount(text: string) {
  const tokenizer = new GPT3Tokenizer({ type: "gpt3" });
  const encoded: { bpe: number[]; text: string[] } = tokenizer.encode(text);
  return encoded.bpe.length;
}

function dotProduct(vecA: Vector, vecB: Vector) {
  let product = 0;
  for (let i = 0; i < vecA.length; i++) {
    product += vecA[i] * vecB[i];
  }
  return product;
}

function magnitude(vec: Vector) {
  let sum = 0;
  for (let i = 0; i < vec.length; i++) {
    sum += vec[i] * vec[i];
  }
  return Math.sqrt(sum);
}

// cosine similarity is the dot products of the two vectors divided by the product of their magnitude
// https://en.wikipedia.org/wiki/Cosine_similarity
function cosineSimilarity(vecA: Vector, vecB: Vector) {
  return dotProduct(vecA, vecB) / (magnitude(vecA) * magnitude(vecB));
}

export function compareEmbeddings(embedding1: Vector, embedding2: Vector) {
  return cosineSimilarity(embedding1, embedding2);
}

// return as many bits as can fit the number of tokens
function getMaxBits(bits: Bit[], maxTokens: number): Bit[] {
  let totalTokens = 0;
  const includedBits: Bit[] = [];
  for (let i = 0; i < bits.length; i++) {
    const bit = bits[i];
    if (totalTokens + bit.tokenCount > maxTokens) {
      return includedBits;
    }
    totalTokens += bit.tokenCount;
    includedBits.push(bit);
  }
  return includedBits;
}

// given an embedding, find the bits with the most similar embeddings
export function getSimilarBits(
  embedding: Vector,
  bits: Bit[],
  maxTokens: number = DEFAULT_TOKEN_COUNT
): Bit[] {
  const sortedBits = bits
    .map((bit) => {
      return {
        ...bit,
        similarity: compareEmbeddings(embedding, bit.embedding),
      };
    })
    // sort by similarity descending
    .sort((a, b) => b.similarity - a.similarity);

  return getMaxBits(sortedBits, maxTokens);
}

// create a single embedding from the given input
// each input must not exceed 8192 tokens in length
// TODO: check token length
export async function generateEmbedding(input: string) {
  const response = await openai.createEmbedding({
    model: "text-embedding-ada-002",
    input: input,
  });

  return response.data.data[0].object as unknown as Vector;
}

// batch create embeddings from the given inputs
export async function generateEmbeddings(inputs: string[]) {
  const response = await openai.createEmbedding({
    model: "text-embedding-ada-002",
    input: inputs,
  });

  return response.data.data.map((item) => item.embedding);
}

export async function craftPrompt(
  args: TemplateArgs,
  template: PromptTemplate,
  bits: Bit[]
) {
  // create an embedding for the given input
  const queryEmbedding = await generateEmbedding(args.input);

  // use that embedding to find the most similar other embeddings as context
  const similarBits = getSimilarBits(queryEmbedding, bits, 1500);

  return template(args, similarBits);
}

export async function craftMessages(
  args: TemplateArgs,
  template: MessagesTemplate,
  bits: Bit[]
) {
  // create an embedding for the given input
  const queryEmbedding = await generateEmbedding(args.input);

  // use that embedding to find the most similar other embeddings as context
  const similarBits = getSimilarBits(queryEmbedding, bits, 1500);

  return template(args, similarBits);
}

export async function createCompletion(prompt: string) {
  // TODO: make all this configurable
  const response = await openai.createCompletion({
    model: "text-davinci-003",
    prompt,
    temperature: 0,
    max_tokens: MAX_COMPLETION_TOKENS,
  });
  return response.data.choices[0].text;
}

export async function createChatCompletion(messages: Message[]) {
  // TODO: make all this configurable
  const response = await openai.createChatCompletion({
    model: "gpt-3.5-turbo",
    messages,
    temperature: 0,
    max_tokens: MAX_COMPLETION_TOKENS,
  });
  return response.data.choices[0].message?.content;
}

// craft dynamic messages with context and send it to OpenAI for completion
export async function createContextualChatCompletion(
  args: TemplateArgs,
  messagesTemplate: MessagesTemplate,
  bits: Bit[]
): Promise<{ messages: Message[]; completion: string | undefined }> {
  const messages = await craftMessages(args, messagesTemplate, bits);
  const completion = await createChatCompletion(messages);
  return { messages, completion };
}

// craft a dynamic prompt with context and send it to OpenAI for completion
export async function createContextualCompletion(
  args: TemplateArgs,
  promptTemplate: PromptTemplate,
  bits: Bit[]
): Promise<{ prompt: string; completion: string | undefined }> {
  const prompt = await craftPrompt(args, promptTemplate, bits);
  const completion = await createCompletion(prompt);
  return { prompt, completion };
}
