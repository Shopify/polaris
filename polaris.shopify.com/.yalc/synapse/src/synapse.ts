/**
 * Synapse
 * Currently only OpenAI is supported
 */

export { defineConfig } from "./defineConfig";

// openAI utils
export {
  encodeEmbedding,
  decodeEmbedding,
  getTokenCount,
  compareEmbeddings,
  generateEmbedding,
  generateEmbeddings,
  craftPrompt,
  craftMessages,
  createCompletion,
  createChatCompletion,
  createContextualChatCompletion,
  createContextualCompletion,
  getSimilarBits,
} from "./openAI";

// plugins
export { queryResponseParse } from "./plugins/queryResponseParse";
export { embeddings } from "./plugins/embeddings";
export { graphQLParse } from "./plugins/graphQLParse";
export { synapseStringify } from "./plugins/synapseStringify";

// types
export type {
  Bit,
  BitNode,
  SynapseTree,
  Vector,
  PromptTemplate,
  TemplateArgs,
  Message,
  MessagesTemplate,
  SynapseConfig,
  SynapseOptions,
} from "./interfaces";
