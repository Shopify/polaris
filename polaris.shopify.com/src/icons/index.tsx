const icons = {
  Polaris: () => (
    <svg width="20" height="20" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M10 17a7 7 0 100-14 7 7 0 000 14zm0 0a7 7 0 017-7 7 7 0 01-7-7 7 7 0 01-7 7 7 7 0 017 7z"
        fill="currentColor"
      />
    </svg>
  ),
  Hexagon: () => (
    <svg
      width="20"
      height="20"
      viewBox="0 0 20 20"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M3 8.09173V11.9079C3 12.5884 3.34604 13.2223 3.91848 13.5903L8.91848 16.8046C9.5773 17.2281 10.4227 17.2281 11.0815 16.8046L16.0815 13.5903C16.654 13.2223 17 12.5884 17 11.9079V8.09173C17 7.4112 16.654 6.77737 16.0815 6.40937L11.0815 3.19508C10.4227 2.77156 9.5773 2.77156 8.91848 3.19508L3.91849 6.40937C3.34604 6.77737 3 7.4112 3 8.09173ZM12 9.99982C12 11.1044 11.1046 11.9998 10 11.9998C8.89543 11.9998 8 11.1044 8 9.99982C8 8.89525 8.89543 7.99982 10 7.99982C11.1046 7.99982 12 8.89525 12 9.99982ZM13.5 9.99982C13.5 11.9328 11.933 13.4998 10 13.4998C8.067 13.4998 6.5 11.9328 6.5 9.99982C6.5 8.06682 8.067 6.49982 10 6.49982C11.933 6.49982 13.5 8.06682 13.5 9.99982Z"
        fill="currentColor"
      />
    </svg>
  ),
};

export default icons;
