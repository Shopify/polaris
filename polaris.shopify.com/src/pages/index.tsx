import HomePage from '../components/HomePage';

const Resources = () => <HomePage />;

export default Resources;
