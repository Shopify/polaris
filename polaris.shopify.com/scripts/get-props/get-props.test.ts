import {getRelevantTypes, getProps, normalizePath} from './src/get-props';
import path from 'path';

it('Normalizes paths', () => {
  expect(
    normalizePath(
      '/Users/martenbjork/my-files/src/polaris/polaris-react/foo.tsx',
    ),
  ).toEqual('polaris-react/foo.tsx');

  expect(normalizePath('/polaris-react/foo.tsx')).toEqual(
    'polaris-react/foo.tsx',
  );

  expect(normalizePath('../polaris-react/foo.tsx')).toEqual(
    'polaris-react/foo.tsx',
  );

  expect(normalizePath('./polaris-react/foo.tsx')).toEqual(
    'polaris-react/foo.tsx',
  );
});

it('Parses interfaces', () => {
  const testFiles = [path.join(__dirname, './testData/interfaces.ts')];
  expect(getProps(testFiles)).toMatchSnapshot();
});

it('Parses type aliases', () => {
  const testFiles = [path.join(__dirname, './testData/types.ts')];
  expect(getProps(testFiles)).toMatchSnapshot();
});

it('Parses enums', () => {
  const testFiles = [path.join(__dirname, './testData/enums.ts')];
  expect(getProps(testFiles)).toMatchSnapshot();
});

it('Gracefully handles unsupported types', () => {
  const testFiles = [path.join(__dirname, './testData/unsupported.ts')];
  expect(getProps(testFiles)).toMatchSnapshot();
});

it('Filters out the relevant AST nodes when given a starting point', () => {
  const testFiles = [
    path.join(__dirname, './testData/enums.ts'),
    path.join(__dirname, './testData/interfaces.ts'),
    path.join(__dirname, './testData/relevantTypes.ts'),
    path.join(__dirname, './testData/types.ts'),
    path.join(__dirname, './testData/unsupported.ts'),
  ];

  const allTypes = getProps(testFiles);
  const filteredTypes = getRelevantTypes(
    allTypes,
    'InterfaceProps',
    'polaris.shopify.com/scripts/get-props/testData/relevantTypes.ts',
  );
  const names = Object.values(filteredTypes).map((node) => node.name);

  expect(names).toMatchSnapshot();
});
