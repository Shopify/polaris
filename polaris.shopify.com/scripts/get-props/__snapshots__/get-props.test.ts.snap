// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters out the relevant AST nodes when given a starting point 1`] = `
Array [
  "InterfaceProps",
  "Relationship",
  "Friend",
  "Family",
  "Close",
  "Extended",
  "Animal",
  "Referenced",
]
`;

exports[`Gracefully handles unsupported types 1`] = `
Object {
  "UnsupportedProps": Object {
    "polaris.shopify.com/scripts/get-props/testData/unsupported.ts": Object {
      "description": "Props that the script can't parse",
      "filePath": "polaris.shopify.com/scripts/get-props/testData/unsupported.ts",
      "name": "UnsupportedProps",
      "syntaxKind": "TypeAliasDeclaration",
      "value": "(Animal | string) & null",
    },
  },
}
`;

exports[`Parses enums 1`] = `
Object {
  "Animal": Object {
    "polaris.shopify.com/scripts/get-props/testData/enums.ts": Object {
      "filePath": "polaris.shopify.com/scripts/get-props/testData/enums.ts",
      "members": Array [
        Object {
          "filePath": "polaris.shopify.com/scripts/get-props/testData/enums.ts",
          "name": "Horse",
          "value": 0,
        },
        Object {
          "filePath": "polaris.shopify.com/scripts/get-props/testData/enums.ts",
          "name": "Dog",
          "value": 10,
        },
        Object {
          "filePath": "polaris.shopify.com/scripts/get-props/testData/enums.ts",
          "name": "Cow",
          "value": "Moo",
        },
      ],
      "name": "Animal",
      "syntaxKind": "EnumDeclaration",
      "value": "export enum Animal {
  Horse,
  Dog = 10,
  Cow = 'Moo',
}",
    },
  },
}
`;

exports[`Parses interfaces 1`] = `
Object {
  "Animal": Object {
    "polaris.shopify.com/scripts/get-props/testData/enums.ts": Object {
      "filePath": "polaris.shopify.com/scripts/get-props/testData/enums.ts",
      "members": Array [
        Object {
          "filePath": "polaris.shopify.com/scripts/get-props/testData/enums.ts",
          "name": "Horse",
          "value": 0,
        },
        Object {
          "filePath": "polaris.shopify.com/scripts/get-props/testData/enums.ts",
          "name": "Dog",
          "value": 10,
        },
        Object {
          "filePath": "polaris.shopify.com/scripts/get-props/testData/enums.ts",
          "name": "Cow",
          "value": "Moo",
        },
      ],
      "name": "Animal",
      "syntaxKind": "EnumDeclaration",
      "value": "export enum Animal {
  Horse,
  Dog = 10,
  Cow = 'Moo',
}",
    },
  },
  "InterfaceProps": Object {
    "polaris.shopify.com/scripts/get-props/testData/interfaces.ts": Object {
      "description": "An interface containing a bunch of complex props",
      "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
      "members": Array [
        Object {
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "[key: string]",
          "value": "any",
        },
        Object {
          "defaultValue": "true",
          "description": "A boolean member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "isBoolean",
          "syntaxKind": "PropertySignature",
          "value": "boolean",
        },
        Object {
          "defaultValue": "Hello",
          "description": "A string member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "stringMember",
          "syntaxKind": "PropertySignature",
          "value": "string",
        },
        Object {
          "defaultValue": "123",
          "description": "A number member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "numberMember",
          "syntaxKind": "PropertySignature",
          "value": "number",
        },
        Object {
          "description": "A method member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "isOptional": true,
          "name": "isMethod",
          "syntaxKind": "PropertySignature",
          "value": "(a: string, b: SomeInterface, c: Animal) => any",
        },
        Object {
          "description": "An enum member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "enumMember",
          "syntaxKind": "PropertySignature",
          "value": "Animal",
        },
        Object {
          "description": "An optional member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "isOptional": true,
          "name": "optionalMember",
          "syntaxKind": "PropertySignature",
          "value": "string",
        },
        Object {
          "deprecationMessage": "This member is deprecated...",
          "description": "A deprecated member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "deprecatedMember",
          "syntaxKind": "PropertySignature",
          "value": "string",
        },
        Object {
          "description": "A member that references another interface",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "referencedInterface",
          "syntaxKind": "PropertySignature",
          "value": "SomeInterface",
        },
        Object {
          "description": "A union member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "unionMember",
          "syntaxKind": "PropertySignature",
          "value": "string | number",
        },
        Object {
          "description": "An intersection member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "intersectionMember",
          "syntaxKind": "PropertySignature",
          "value": "{ a: string; } & { b: string; }",
        },
        Object {
          "description": "An array member",
          "filePath": "polaris.shopify.com/scripts/get-props/testData/interfaces.ts",
          "name": "arrayMember",
          "syntaxKind": "PropertySignature",
          "value": "[1, \\"hello\\", () => void]",
        },
      ],
      "name": "InterfaceProps",
      "value": "interface InterfaceProps {
  /**
   * A boolean member
   *
   * @default true
   */
  isBoolean: boolean;
  /**
   * A string member
   *
   * @default Hello
   */
  stringMember: string;
  /**
   * A number member
   *
   * @default 123
   */
  numberMember: number;
  /**
   * A method member
   */
  isMethod?: (a: string, b: SomeInterface, c: Animal) => any;
  /**
   * An enum member
   */
  enumMember: Animal;
  /**
   * An optional member
   */
  optionalMember?: string;
  /**
   * A deprecated member
   *
   * @deprecated This member is deprecated...
   */
  deprecatedMember: string;
  /**
   * A member that references another interface
   */
  referencedInterface: SomeInterface;
  /**
   * A union member
   */
  unionMember: string | number;
  /**
   * An intersection member
   */
  intersectionMember: {a: string} & {b: string};
  /**
   * An array member
   */
  arrayMember: [1, 'hello', () => void];
  /**
   * Index signature
   */
  [key: string]: any;
}",
    },
  },
}
`;

exports[`Parses type aliases 1`] = `
Object {
  "MyFunction": Object {
    "polaris.shopify.com/scripts/get-props/testData/types.ts": Object {
      "description": "A function type",
      "filePath": "polaris.shopify.com/scripts/get-props/testData/types.ts",
      "name": "MyFunction",
      "syntaxKind": "TypeAliasDeclaration",
      "value": "(a: string, b: number) => void",
    },
  },
  "StringProps": Object {
    "polaris.shopify.com/scripts/get-props/testData/types.ts": Object {
      "description": "A basic type",
      "filePath": "polaris.shopify.com/scripts/get-props/testData/types.ts",
      "name": "StringProps",
      "syntaxKind": "TypeAliasDeclaration",
      "value": "string",
    },
  },
  "TypeProps": Object {
    "polaris.shopify.com/scripts/get-props/testData/types.ts": Object {
      "description": "A more complex type",
      "filePath": "polaris.shopify.com/scripts/get-props/testData/types.ts",
      "name": "TypeProps",
      "syntaxKind": "TypeAliasDeclaration",
      "value": "{
  a: string;
  b: number;
}",
    },
  },
}
`;
